#### (1)-
##### Z test is used when the population standard deviation is known or when the sample size is greater than equal to 30.
##### T test is used when the population standard deviation is not known. 
##### Ex) We will use Z test here- Let’s say we need to determine if girls on average score higher than 600 in the exam. We have the information that the standard deviation for girls’ scores is 100. So, we collect the data of 20 girls by using random samples and record their marks. Finally, we also set our ⍺ value (significance level) to be 0.05. 
##### We have population standard deviation apart from population mean, sample mean, so we use Z test.
##### For the same ex) Let’s say we want to determine if on average girls score more than 600 in the exam. We do not have the information related to variance (or standard deviation) for girls’ scores. To a perform t-test, we randomly collect the data of 10 girls with their marks and choose our ⍺ value (significance level) to be 0.05 for Hypothesis Testing.
##### Here we have population mean, sample mean and sample standard deviation but not population standard deviation, so we use T test.

#### (2)-
##### The one-tailed test refers to a test of null hypothesis, in which the alternative hypothesis is articulated directionally. Here, the critical region lies only on one tail. 
##### However, if the alternative hypothesis is not exhibited directionally, then it is known as the two-tailed test of the null hypothesis, wherein the critical region is on both the tails. 
##### One-tailed tests will only have one critical region whereas two-tailed tests will have two critical regions.

#### (3)-
##### Type 1 error- We reject the Null hypothesis when in reality it is True.

##### Type 2 error- We accept the Null hypothesis when in reality it is False.
##### Ex) A man undergoes trail for commiting murder of his wife. Here Null hypothesis H0- Man is not guilty. Alternative hypothesis H1- Man is guilty.
##### Type 1 error- We reject the Null hypothesis when in reality it is True so, the man is charged for murder when in reality he hasn't commited murder and is not guilty.
##### Type 2 error- We accept the Null hypothesis when in reality it is False so, the man is not charged for murder i.e. not guilty when in reality he has murdered and is guilty.

#### (4)-
##### Bayes's theorem- The Bayes theorem is a mathematical formula for calculating conditional probability in probability and statistics. In other words, it's used to figure out how likely an event is based on its proximity to another. 
##### P(B|A)= (P(B)* P(A|B))/P(A)
##### where P(A|B) is the probability of A occuring, given B is True. P(B) and P(A) are independent probabilities of B and A respectively.
##### Ex) A bag A contains 1 white and 6 red balls. Another bag B contains 4 white and 3 red balls. One of the bags is selected at random and a ball is drawn from it, which is found to be white. Find the probability that the ball drawn is from the bag A.
##### let the events be E1= choosing bag A, E2= choosing bag B, E= choosing white ball.
##### P(E1)= 1/2, P(E2)= 1/2
##### P(E/E1)= 1/7, P(E/E2)= 4/7
##### We have to find the probability that the white ball drawn is from bag A- P(E1/E) = (P(E1) * P(E/E1)) / (P(E1)* P(E/E1) + P(E2) * P(E/E2))
##### P(E1/E)= (1/2 * 1/7)/ (1/2 * 1/7 + 1/2 * 4/7) = 1/5

#### (5)-
##### A confidence interval refers to the interval within which a population parameter will fall for a certain proportion of times. Confidence interval comes into picture because point estimate is not a perfect estimate for unknown population parameter. So, we use interval estimate and this interval is the confidence interval. Confidence interval = point estimate +- (margin of error)

In [2]:
# example of calculating confidence interval-
# The sample mean CAT score is 520 for 80 test takers with a standard deviation of 100. 
# Estimate population mean with 95% confidence interval

import numpy as np
sample_mean= 520
sample_std= 100
n= 80             # this is sample size
CI= 0.95
significance_level= 1-CI

# Z score from Z table for significance level 0.05 i.e AUC= 0.025 is-
z= 1.96         # i.e z = +- 1.96 

# calculating margin of error-
MOE= 1.96* (sample_std/np.sqrt(n))

# calculating lower and upper bounds of confidence interval within which population mean will lie, we are 95% confident about it
lower_bound= sample_mean - MOE
print("lower bound of confidence interval is", lower_bound)
upper_bound= sample_mean + MOE
print("upper bound of confidence interval is", upper_bound)

lower bound of confidence interval is 498.08653382050204
upper bound of confidence interval is 541.913466179498


#### (6)-
##### sample problem- An insurance company insured 2000 scooter drivers, 4000 car drivers, and 6000 truck drivers. The probability of an accident involving a scooter driver, car driver, and a truck driver is 0.01, 0.03, and 0.15 respectively. One of the insured persons meets with an accident. What is the probability that he is a scooter driver?

##### using Bayes's theorem we will solve this- let us define the events as- 
##### E1= person is scooter driver
##### E2= person is car driver
##### E3= person is truck driver
##### A= person meets with accident
##### Now, probability of insured person being scooter driver is P(E1)= 2000/12000= 1/6 
##### Probability of insured person being car driver is P(E2)= 4000/12000= 1/3
##### Probability of insured person being truck driver is P(E3)= 6000/12000= 1/2
##### Given, Probability of accident involving scooter driver is P(A|E1)= 0.01
##### Given, Probability of accident involving car driver is P(A|E2)= 0.03
##### Given, Probability of accident involving truck driver is P(A|E3)= 0.15
##### We have to find the probability of a scooter driver being in accident i.e. accident has occured and what is the probability of that person being a scooter driver P(E1|A)= P(E1)* P(A|E1)/(P(E1)* P(A|E1) + P(E2)* P(A|E2) + P(E3)* P(A|E3))
##### P(E1|A)= (1/6* 0.01)/(1/6* 0.01 + 1/3* 0.03 + 1/2* 0.15) = 0.001666/0.019156 = 0.0192

#### (7)-

In [3]:
sample_mean= 50
sample_std= 5
n= 30
CI= 0.95
significance_level= 0.05

# Z score from Z table for significance level 0.05 i.e AUC= 0.025 is-
z= 1.96         # i.e z = +- 1.96 

# calculating margin of error-
MOE= 1.96* (sample_std/np.sqrt(n))

# calculating lower and upper bounds of confidence interval within which population mean will lie, we are 95% confident about it
lower_bound= sample_mean - MOE
print("lower bound of confidence interval is", lower_bound)
upper_bound= sample_mean + MOE
print("upper bound of confidence interval is", upper_bound)

lower bound of confidence interval is 48.21077297881646
upper bound of confidence interval is 51.78922702118354


#### (8)-
##### Margin of error is nothing but the confidence interval and it extends equally on both sides of the mean. So,      2* margin of error = confidence interval. A margin of error tells you how many percentage points your results will differ from the real population parameter. The smaller the margin of error, the more confidence we may have in our results. The bigger the margin of error, the farther they can stray from the views of the total population. 
##### Greater the sample size n, smaller will be the margin of error because MOE= Z-score * std/sqrt(n). So, they share an inverse relationship.

In [5]:
# Example- 
n1= 30
sample_mean= 50
sample_std= 5
CI= 0.95
significance_level= 0.05

## Z score from Z table for significance level 0.05 i.e AUC= 0.025 is-
z= 1.96         # i.e z = +- 1.96 

#calculating margin of error-
MOE1= 1.96* (sample_std/np.sqrt(n1))
print("margin of error is {} when sample size is {}".format(MOE1, n1))

# taking a bigger sample size and keeping rest all same-
n2= 100

#calculating margin of error-
MOE2= 1.96* (sample_std/np.sqrt(n2))
print("margin of error is {} when sample size is {}".format(MOE2, n2))

margin of error is 1.7892270211835426 when sample size is 30
margin of error is 0.98 when sample size is 100


#### (9)-

In [6]:
x= 75
population_mean= 70
population_std= 5
z_score= (x-population_mean)/population_std
print("the z score for the data point {} is {}".format(x, z_score))

the z score for the data point 75 is 1.0


##### z_score of 1 signifies that how many standard devaitions away from the mean that point lies. So, the point 75 is 1 standard deviation from the mean.

#### (10)-
##### Null hypothesis(H0)- The drug is not effective i.e. no weight loss so, mean= 0
##### Alternate hypothesis(H1)- The drug is effective i.e. weight loss happens so, mean != 0 It is observed that the mean weight loss is 6 pounds. We will use 2 tail T test.

In [9]:
import numpy as np
population_mean= 0
sample_mean= 6
sample_std= 2.5
n= 50
CI= 0.95
significance_value= 0.05
df= n-1            # this is the degree of freedom

# calculating the critical values on both tails-
# corresponding to significance value 0.05, and df= 49 in the 2 tail t test table, we get-
t1= 2.0096    # i.e. t1= +- 2.0096

# calculating t value from t stats-
t2= (sample_mean-population_mean)/(sample_std/np.sqrt(n))
print("t score from t stats", t2)

# checking if we accept or reject null hypothesis-
if t2 > t1:
    print("Reject Null hypothesis")
else:
    print("Fail to reject Null hypothesis")

t score from t stats 16.970562748477143
Reject Null hypothesis


##### So, the new drug worked and the average loss in weight is 6 pounds

#### (11)-
##### As it is a proportion problem, so the distribution is binomial. p is the probability of success and q=1-p is the probablity of failure.

In [11]:
n= 500       # this is the sample size
CI= 0.95
significance_level= 1-CI
# the no. of people satisfied with their current job is 65% of n-
x= 0.65*n 

# the estimated proportion of successes i.e. probability of success-
p_prime= x/n
q_prime= 1-p_prime      # probability of failure

# Z score from z table corresponding to significance level = 0.05 i.e. AUC= 0.025 is 
z= 1.96       # i.e. z= +-1.96

# calculating error bound for proportion-
EBP= z* (np.sqrt(p_prime*q_prime)/np.sqrt(n))
print("Error bound for proportion is", EBP)

# calculating lower and upper bound of confidence interval for true proportion-
lower_bound= p_prime - EBP
print("lower bound of confidence interval is", lower_bound)
upper_bound= p_prime + EBP
print("upper bound of confidence interval is", upper_bound)

Error bound for proportion is 0.04180822885509502
lower bound of confidence interval is 0.608191771144905
upper bound of confidence interval is 0.6918082288550951


##### so, the confidence interval for the true proportion of people who are satisfied with their job is from 0.608 to 0.692, we are 95% confident about it.

#### (12)-
##### we will use a 2 sample T test. We can use that because the variances of the 2 groups are similar i.e. the ratio of variances is less than 4:1. We have to compare if the 2 teaching methods have a significant difference in student performance i.e. we compare the means of the two groups.
##### Null hypothesis(H0)- The means of 2 groups are equal i.e. mean1= mean2
##### Alternate hypothesis(H1)- The means of 2 groups are not equal i.e. mean1 != mean2

In [12]:
# let us consider sample size = 30 for both groups
import numpy as np
n1= 30
n2= 30
sample_mean1= 85
sample_std1= 6
sample_mean2= 82
sample_std2= 5
significance_level= 0.01

# calculating difference of averages-
diff_of_means= sample_mean1 - sample_mean2

# calculating pooled variance and standard deviation-
pooled_var= (sample_std1**2 + sample_std2**2)/2
pooled_std= np.sqrt(pooled_var)

# calculating t score of 2 sample t test stats-
t= diff_of_means/ (pooled_std * np.sqrt((1/n1) + (1/n2)))
print("the t value of 2 sample t test is", t)

the t value of 2 sample t test is 2.10386061995483


In [13]:
tot_df= (n1-1)+(n2-1)
tot_df

58

In [14]:
# using the total degree of freedom and significance level for 2 tailed test in the t table we get t value as-
t_table_value= 2.6633

# checking if we accept or reject null hypothesis- 
if t > t_table_value:
    print("Reject Null hypothesis")
else:
    print("Fail to reject Null hypothesis")

Fail to reject Null hypothesis


##### So, the t value lies in the acceptance area and we fail to reject the null hypothesis, thus the means of 2 groups are equal and there is no significant difference in student performance for the 2 methods of teaching

#### (13)-

In [15]:
import numpy as np
population_mean= 60
population_std= 8
sample_mean= 65
n= 50         # this is sample size
CI= 0.90
significance_level= 1-CI

# z score from z table corresponding to significance level = 0.10 i.e. AUC= 0.05 is- 
z= 1.64         # i.e. z= +- 1.64

# calculating margin of error-
MOE= z* (population_std/np.sqrt(n))
print("The margin of error is", MOE)

# calculating lower and upper bounds of confidence interval within which true population mean will lie, we are 90% confident about it-
lower_bound= sample_mean - MOE
print("lower bound of confidence interval is", lower_bound)
upper_bound= sample_mean + MOE
print("upper bound of confidence interval is", upper_bound)

The margin of error is 1.8554481938335006
lower bound of confidence interval is 63.1445518061665
upper bound of confidence interval is 66.8554481938335


#### (14)-
##### Null hypothesis(H0)- The caffiene has no significant effect on reaction time i.e. mean = 0
##### Alternate hypothesis(H1)- caffiene has effect on reaction time i.e mean != 0 It is observed that the average reaction time was 0.25 seconds.
##### We conduct a 2 tailed t test

In [16]:
import numpy as np
population_mean= 0
n= 30
sample_mean= 0.25
sample_std= 0.05
CI= 0.90
significance_level= 1-CI
df= n-1

# from t table (2 tailed), the t value corresponding to significance level = 0.10 and degree of freedom = 29 is-
t_table_value= 1.699        # i.e. t = +- 1.699

# calculating t value from t stats-
t= (sample_mean - population_mean)/(sample_std/np.sqrt(n))
print("t value from t stats is", t)

# checking if we accept or reject null hypothesis-
if t > t_table_value:
    print("Reject Null hypothesis")
else:
    print("Fail to reject Null hypothesis")

t value from t stats is 27.386127875258307
Reject Null hypothesis


##### The t value from t stats lies in the rejection area, so Null hypothesis is rejected which means caffeine has significant effect on reaction time and it is 0.25 seconds.